kubernetes/schema_affinity_spec.go

```diff
@@ -0,0 +1,185 @@
+						Optional:    true,
+						Elem:        &schema.Schema{Type: schema.TypeString},
+						Set:         schema.HashString,
+					},
+				},
+			},
+		},
+	}
+}
+
+func podAffinityFields() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"required_during_scheduling_ignored_during_execution": {
+			Type:        schema.TypeList,
+			Description: "If the affinity requirements specified by this field are not met at scheduling time, the pod will not be scheduled onto the node. If the affinity requirements specified by this field cease to be met at some point during pod execution (e.g. due to a pod label update), the system may or may not try to eventually evict the pod from its node. When there are multiple elements, the lists of nodes corresponding to each PodAffinityTerm are intersected, i.e. all terms must be satisfied.",
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: podAffinityTermFields(),
+			},
+		},
+		"preferred_during_scheduling_ignored_during_execution": {
+			Type:        schema.TypeList,
+			Description: "The scheduler will prefer to schedule pods to nodes that satisfy the anti-affinity expressions specified by this field, but it may choose a node that violates one or more of the expressions. The node that is most preferred is the one with the greatest sum of weights, i.e. for each node that meets all of the scheduling requirements (resource request, RequiredDuringScheduling anti-affinity expressions, etc.), compute a sum by iterating through the elements of this field and adding 'weight' to the sum if the node matches the corresponding MatchExpressions; the node(s) with the highest sum are the most preferred.",
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: weightedPodAffinityTermFields(),
+			},
+		},
+	}
+}
+
+func podAffinityTermFields() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"label_selector": {
+			Type:        schema.TypeList,
+			Description: "A label query over a set of resources, in this case pods.",
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: labelSelectorFields(false),
+			},
+		},
+		"namespaces": {
+			Type:        schema.TypeSet,
+			Description: "namespaces specifies which namespaces the labelSelector applies to (matches against); null or empty list means 'this pod's namespace'",
+			Optional:    true,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Set:         schema.HashString,
+		},
+		"topology_key": {
+			Type:        schema.TypeString,
+			Description: "empty topology key is interpreted by the scheduler as 'all topologies'",
+			Optional:    true,
+		},
+	}
+}
+
+func weightedPodAffinityTermFields() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"weight": {
+			Type:        schema.TypeInt,
+			Description: "weight associated with matching the corresponding podAffinityTerm, in the range 1-100",
+			Required:    true,
```

Comment:
Same thing about validation.
```suggestion
			Required:    true,
			ValidateFunc: validation.IntBetween(1, 100),
```